services/worker/src/worker/loop.py

```diff
@@ -125,11 +125,20 @@ def process_next_job(self) -> bool:
         logging.debug("try to process a job")
 
         try:
-            job_info = self.queue.start_job(only_job_types=self.worker_config.only_job_types)
-            if self.worker_config.only_job_types and job_info["type"] not in self.worker_config.only_job_types:
+            job_info = self.queue.start_job(
+                job_types_blocked=self.worker_config.job_types_blocked,
+                job_types_only=self.worker_config.job_types_only,
+            )
+            if self.worker_config.job_types_blocked and job_info["type"] in self.worker_config.job_types_blocked:
```

Comment:
Do we really need this validation twice? One here and the other in queue start which is called in process_next_job -> run

Reply:
I think we can remove one, you're right